### ISSUE TITLE
Insert linebreak shortcut

### ISSUE THREAD

**Issue by Greedquest** [rubberduck-vba/Rubberduck#5131]
**Justification**
When typing multiline code (such as long function signatures), it would be nice to be able to leverage auto-indenting line-breaks the same way the smart-concat feature does.

**Description**
Alt+Enter might be a good candidate; it's what Excel uses for line breaks within cells, and it doesn't clash with existing smart-concat stuff.

### Current:
```vb
Private Declare Function PeekMessage Lib "user32" Alias "PeekMessageA" (|
```
hitting <kbd>_</kbd> then <kbd>Enter</kbd> gives:

```vb
Private Declare Function PeekMessage Lib "user32" Alias "PeekMessageA" ( _
| ':(
```
### Proposed:

With the same setup as above, instead hit <kbd>Alt</kbd> + <kbd>Enter</kbd> to give:
```vb
Private Declare Function PeekMessage Lib "user32" Alias "PeekMessageA" ( _
                         | 'Caret right where you want to type :)
```

**Additional Context**

I'd expect the behaviour mid string to be
```vb
myString = "Here's another long string|"
```
<kbd>Alt</kbd> + <kbd>Enter</kbd>
```vb
myString = "A long string _
          |" 'or whatever the indenter usually does in this situation
```
...producing non-compiling code but consistent user experience

**Comment by retailcoder** [comment]
Some of this needs quite a bit of mind-reading that Rubberduck doesn't do. Also I don't understand the `myString` example, which clearly *does* interfere with smart-concat.

**Comment by Greedquest** [comment]
@retailcoder Sorry I wasn't clear:

Smart-concat uses `Enter` & `Ctrl+Enter` I believe, for min-string line breaks. The proposed feature is a similar shortcut for general (typically out of string) line breaks. 

I included that last example to illustrate what might happen if you use `Alt+Enter`/ the proposed feature *within in a string*; namely that it would not do any clever string splitting stuff like smart-concat does, it would just insert the line continuation and attempt to indent the (now broken) code. I.e. you wouldn't do that - smart-concat would remain unchanged and so for nice line continuations within strings just use that.

Also just to be clear, Alt+Enter doesn't already exist as a keyboard shortcut in the VBE, it's just what Excel uses to add newlines to formulae as you type.

I hope that clears it up; I think with that there shouldn't be any mind reading necessary - the proposed feature does the same thing in all contexts, just adds the line break and puts the caret at the logical indentation level on the next line.

**Comment by retailcoder** [comment]
TBH I was reserving <kbd>Alt</kbd>+<kbd>Enter</kbd> for a future "quickfix" or similar command, when (not if!) inspection results show up directly in the code pane. R# uses that shortcut for that purpose.

**Comment by Greedquest** [comment]
Also with regard to the indenter not handling multi line declare statements, I'm not sure what you mean, but for me it turns this:

```vb
Private Declare Function PeekTimerMessage Lib "user32" Alias "PeekMessageA" ( _
ByRef outMessage As timerMessage, _
ByVal hWnd As LongPtr, _
Optional ByVal wMsgFilterMin As WindowsMessage = WM_TIMER, _
Optional ByVal wMsgFilterMax As WindowsMessage = WM_TIMER, _
Optional ByVal wRemoveMsg As PeekMessageFlag = PM_REMOVE) As Long
```
into this
```
Private Declare Function PeekTimerMessage Lib "user32" Alias "PeekMessageA" ( _
                         ByRef outMessage As timerMessage, _
                         ByVal hWnd As LongPtr, _
                         Optional ByVal wMsgFilterMin As WindowsMessage = WM_TIMER, _
                         Optional ByVal wMsgFilterMax As WindowsMessage = WM_TIMER, _
                         Optional ByVal wRemoveMsg As PeekMessageFlag = PM_REMOVE) As Long
```
So when I said "right where you want to type", I just meant wherever the next line would normally get indented to  - in this case the `P` of `PeekTimerMessage`. Or did you mean that the level of indentation depends on what gets typed next? Perhaps you're referring to the fact that the indenter fails for `PtrSafe` declarations.

Also I'm not limiting this to function signatures, a complex addition or concatenation too could benefit:

        GetGUID = _
                String(8 - Len(Hex$(udtGUID.Data1)), "0") & Hex$(udtGUID.Data1) _
                & String(4 - Len(Hex$(udtGUID.Data2)), "0") & Hex$(udtGUID.Data2) _
                & String(4 - Len(Hex$(udtGUID.Data3)), "0") & Hex$(udtGUID.Data3) _
                & IIf((udtGUID.Data4(0) < &H10), "0", "") & Hex$(udtGUID.Data4(0)) _
                & IIf((udtGUID.Data4(1) < &H10), "0", "") & Hex$(udtGUID.Data4(1)) _
                & IIf((udtGUID.Data4(2) < &H10), "0", "") & Hex$(udtGUID.Data4(2)) _
                & IIf((udtGUID.Data4(3) < &H10), "0", "") & Hex$(udtGUID.Data4(3)) _
                & IIf((udtGUID.Data4(4) < &H10), "0", "") & Hex$(udtGUID.Data4(4)) _
                & IIf((udtGUID.Data4(5) < &H10), "0", "") & Hex$(udtGUID.Data4(5)) _
                & IIf((udtGUID.Data4(6) < &H10), "0", "") & Hex$(udtGUID.Data4(6)) _
                & IIf((udtGUID.Data4(7) < &H10), "0", "") & Hex$(udtGUID.Data4(7))

**Comment by retailcoder** [comment]
> Perhaps you're referring to the fact that the indenter fails for PtrSafe declarations.

Being on a 64-bit host, that would be it. Is there an issue for this indenter bug?

**Comment by Greedquest** [comment]
@retailcoder Yup: #4795

**Comment by retailcoder** [comment]
TBH I'm thinking this issue will be moot once block-AC is implemented - the plan is to ship "auto-indent" along with block completion.

**Comment by CHR-IS-B** [comment]
> Alt+Enter was only a suggestion because it might be familiar

Right!  For us Excel users, when typing inside a cell, ALT + ENTER adds a line break inside the cell without advancing to the next cell, but another shortcut would work just as well.

**Comment by retailcoder** [comment]
Note that we're talking *defaults* here - all hotkeys are fully configurable.

**Comment by CHR-IS-B** [comment]
Gotta love configurable hotkeys!
